src/auth/dto/create-user.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  IsString,
+  MinLength,
+  Matches,
+  IsNotEmpty,
+  IsEmail,
+  Length,
+} from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+export class CreateUserDto {
+  @IsString()
+  @ApiProperty({
+    example: 'username',
+  })
+  @IsNotEmpty({ message: 'First name is required' })
+  name: string;
+
+  @Length(8, 18, {
+    message: 'Phone number should be between 8 and 18 digits',
+  })
+  // @Matches(/^\+\d{1,3}-?\d{7,15}$/, {
+  //   message:
+  //     'Phone number should start with a + and country code (1 to 3 digits), followed by an optional dash, and then 7 to 15 digits',
```

Comment:
PullProofAI

Consider uncommenting the validation rule for the phone number to enforce the specified format.
```diff
+   // @Matches(/^\+\d{1,3}-?\d{7,15}$/, {
+   //   message:
+   //     'Phone number should start with a + and country code (1 to 3 digits), followed by an optional dash, and then 7 to 15 digits',
+   // })
```


<!-- This is an auto-generated comment by PullProofAI -->